src/codec.rs

```diff
@@ -125,78 +154,126 @@ fn number_of_digits(mut n: usize) -> usize {
     num_digits
 }
 
-impl<T: DeserializeOwned> Decoder for LanguageServerCodec<T> {
-    type Item = T;
+impl<T: DeserializeOwned + std::fmt::Debug> Decoder for LanguageServerCodec<T> {
```

Comment:
Since we already have `std::fmt::Display` imported from up above, perhaps we could also import `Debug` as well for consistency?

Reply:
Oh, whoops, that was actually something I had in place for testing that I forgot to remove.

Reply:
No worries! I do see that the `std::fmt::Debug` nit hasn't been addressed yet, though?

Reply:
Ah, an oversight. Fixed now.